app/models/notifications/mentioned.rb

```diff
@@ -13,19 +13,26 @@ def deleted_translation_key
     end
 
     def linked_object
-      target.post
+      target.mentions_container
     end
 
     def self.notify(mentionable, recipient_user_ids)
       actor = mentionable.author
+      relevant_mentions = filter_mentions(
+        mentionable.mentions.local.where.not(person: mentionable.author),
+        mentionable,
+        recipient_user_ids
+      )
 
-      mentionable.mentions.select {|mention| mention.person.local? }.each do |mention|
+      relevant_mentions.each do |mention|
         recipient = mention.person
-
-        next if recipient == actor || !(mentionable.public || recipient_user_ids.include?(recipient.owner_id))
-
         create_notification(recipient.owner_id, mention, actor).email_the_user(mention, actor)
       end
     end
+
+    def self.filter_mentions(mentions, mentionable, recipient_user_ids)
```

Comment:
I think renaming `Mentioned` to `MentionedInPost` in order to free the name for the shared module is fair.